React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,657 @@
+```
+
+### Colors
+
+Color naming should a single value or an object with keys in range from 50 to 900.
+
+Example:
+
+```ts
+export const colors = {
+  white: "#ffffff",
+  primary: {
+    50: "#f6faff",
+    100: "#d7e9fd",
+    200: "#c0dcfc",
+    300: "#a7cefb",
+    400: "#8abdfa",
+    500: "#68aaf8",
+    600: "#3d92f6",
+    700: "#0070f3",
+    800: "#0064d8",
+    900: "#0055b9",
+  },
+};
+```
+
+### Pseudo props
+
+Pseudo props in ChakraUI can be passed as props to component. Full list of props can be found in [docs](https://chakra-ui.com/docs/features/style-props#pseudo)
+
+Example:
+
+```tsx
+<Button
+  bg="blue.100"
+  _hover={{
+    background: "white",
+  }}
+>
+  Submit
+</Button>
+```
+
+Also pseudo elements can be used in theme or in chakra factory:
+
+```tsx
+const Card = chakra("div", {
+  baseStyle: {
+    bg: "blue.100",
+    _hover: {
+      bg: "white",
+    },
+  },
+});
+
+// or
+const Card = chakra("div", {
+  baseStyle: {
+    bg: "blue.100",
+    ":hover": {
+      bg: "white",
+    },
+  },
+});
+```
+
+### Global styles
+
+Global styles are theme-aware styles you can apply to any html element globally.
+
+They are defined in `src/styles/theme/styles.ts` file.
+
+Example:
+
+```ts
+const overrides = {
+  styles: {
+    global: {
+      body: {
+        fontFamily: "body",
+        bg: "white",
+        color: "primary.200",
+      },
+      "*": {
+        boxSizing: "border-box",
+      },
+    },
+  },
+};
+```
+
+### Responsive
+
+Chakra UI supports responsive styles out of the box. Instead of manually adding @media queries and adding nested styles throughout your code, Chakra UI allows you to provide object and array values to add mobile-first responsive styles.
+
+> Under the hood `@media(min-width)` media query is used to ensure mobile-first
+
+Chakra UI default breakpoints:
+
+```ts
+export const breakpoints = {
+  sm: "30em",
+  md: "48em",
+  lg: "62em",
+  xl: "80em",
+};
+```
+
+Here's how to interpret this syntax:
+
+- `base`: From `0em` upwards
+- `md`: From `48em` upwards
+- `lg`: From `62em` upwards
+- `xl`: From `80em` upwards
+
+For responsive styles, array or object syntax can be used.
+
+```tsx
+// array syntax
+<Text fontSize={["24px", "40px", "56px"]}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+```tsx
+// object syntax
+<Text fontSize={{ base: "24px", md: "40px", lg: "56px" }}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+In case we need to skip a certain breakpoint, `null` is passed at that position in the array to avoid generating unnecessary CSS.
+
+Example:
+
+```jsx
+<Text fontSize={["24px", null, "56px"]}>Lorem Ipsum is simply dummy text</Text>
+```
+
+Array and Object syntax work for every style prop in the theme specification, which means you can change most properties at a given breakpoint. Preferred way is to use array.
+
+To create custom breakpoints read [docs](https://chakra-ui.com/docs/features/responsive-styles#customizing-breakpoints).
```

Comment:
I had some issue with this while our project `tsconfig` had `strict: false`. We should add a note that in order for everything to work as expected in `chakra-ui@1.3.2` and later you need to have `strict: true` in your `tsconfig` otherwise you will get a type error that `breakpoint.toString` are incompatible.

Here is the sandbox of the error: https://codesandbox.io/s/sleepy-sanne-t3fk4